### PR TITLE
Yotam dev2

### PR DESCRIPTION
**Pull request by ioorik**

### CODE REVIEW

File: Plane.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import pygame
```
Comment (mmaylat) [inline Plane.py]: Please complete documentation for all modules

File: bases/PlaneBase.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        self.gridSizeX: int = gridSizeX
+        self.gridSizeY: int = gridSizeY
+        if gridSizeZ is not None:
+            self.gridSizeZ = gridSizeZ
+
+    @abc.abstractmethod
+    def draw(self):
+        """Draw the world/plane to the given surface"""
+
+    def addShape(self, shape: IShape):
```
Comment (mmaylat) [inline bases/PlaneBase.py]: This i puzzling, either should change to:
```suggestion
    def add(self, shape: IShape):
```
Or the interface method should rename to `addShape`

**Review by mmaylat — CHANGES_REQUESTED** [review]
Please see my comment, in addition:
- The CI is broken please make all is passing
- The current commit granularity/description is meaningless to me, please squash/reorder/etc.
- Please rebase on top of the master in advance

File: Plane.py (hunk 1 above)
Comment (ioorik) [reply Plane.py]: Done

File: bases/PlaneBase.py (hunk 2 above)
Comment (ioorik) [reply bases/PlaneBase.py]: Done, with a few changes

**Review by mmaylat — APPROVED** [review]

**Pull request merged**